test/boost/mvcc_test.cc

```diff
@@ -863,6 +863,410 @@ SEASTAR_TEST_CASE(test_partition_snapshot_row_cursor) {
+                BOOST_REQUIRE(cur.continuous());
+
+                BOOST_REQUIRE(cur.next());
+                BOOST_REQUIRE(eq(cur.table_position(), table.make_ckey(3)));
+                BOOST_REQUIRE(!cur.continuous());
+
+                BOOST_REQUIRE(cur.next());
+                BOOST_REQUIRE(eq(cur.table_position(), table.make_ckey(1)));
+                BOOST_REQUIRE(cur.continuous());
+
+                BOOST_REQUIRE(cur.next());
+                BOOST_REQUIRE(eq(cur.table_position(), table.make_ckey(0)));
+                BOOST_REQUIRE(cur.continuous());
+
+                BOOST_REQUIRE(!cur.next());
+            }
+        });
+    });
+}
+
+SEASTAR_TEST_CASE(test_ensure_entry_in_latest_in_reversed_mode) {
+    return seastar::async([] {
+        cache_tracker tracker;
+        auto& r = tracker.region();
+        with_allocator(r.allocator(), [&] {
+            simple_schema table;
+            auto&& s = *table.schema();
+
+            auto e = partition_entry::make_evictable(s, mutation_partition(table.schema()));
+            auto snap1 = e.read(r, tracker.cleaner(), table.schema(), &tracker);
+
+            {
+                auto&& p1 = snap1->version()->partition();
+                p1.clustered_row(s, table.make_ckey(3), is_dummy::no, is_continuous::no);
+                p1.clustered_row(s, table.make_ckey(5), is_dummy::no, is_continuous::no);
+                p1.ensure_last_dummy(s);
+            }
+
+            auto snap2 = e.read(r, tracker.cleaner(), table.schema(), &tracker, 1);
+
+            {
+                auto&& p2 = snap2->version()->partition();
+                p2.clustered_row(s, table.make_ckey(1), is_dummy::no, is_continuous::yes);
+                p2.clustered_row(s, table.make_ckey(5), is_dummy::no, is_continuous::no);
+                p2.ensure_last_dummy(s);
+            }
+
+            auto rev_s = s.make_reversed();
+            partition_snapshot_row_cursor cur(*rev_s, *snap2, false, true);
+            position_in_partition::equal_compare eq(s);
+
+            {
+                logalloc::reclaim_lock rl(r);
+                BOOST_REQUIRE(cur.advance_to(table.make_ckey(3)));
+                BOOST_REQUIRE(eq(cur.table_position(), table.make_ckey(3)));
+                BOOST_REQUIRE(!cur.continuous());
+
+                {
+                    auto res = cur.ensure_entry_in_latest();
+                    BOOST_REQUIRE(res.inserted);
+                    BOOST_REQUIRE(eq(res.row.position(), table.make_ckey(3)));
+                }
+
+                BOOST_REQUIRE(cur.advance_to(table.make_ckey(3)));
+                BOOST_REQUIRE(!cur.continuous());
+
+                {
+                    auto res = cur.ensure_entry_in_latest();
+                    BOOST_REQUIRE(!res.inserted);
+                }
+            }
+        });
+    });
+}
+
+SEASTAR_TEST_CASE(test_ensure_entry_in_latest_does_not_set_continuity_in_reversed_mode) {
+    return seastar::async([] {
```

Comment:
NIT: SEASTAR_THREAD_TEST_CASE?

Reply:
I'll blanket convert the whole file after this PR, ok?